pkg/flare/archive.go

```diff
@@ -84,18 +88,54 @@ type filePermsInfo struct {
 	group string
 }
 
+// Profile is a performance profile containing heap and CPU profiles.
+type Profile struct {
+	FirstHeapProfile  []byte
+	SecondHeapProfile []byte
+	CPUProfile        []byte
+}
+
+// CreatePerformanceProfile creates a CPU and Heap profile to include
+func CreatePerformanceProfile(profileDuration int) (*Profile, error) {
+	cpuProfURL := fmt.Sprintf("http://127.0.0.1:%s/debug/pprof/profile?seconds=%d", config.Datadog.GetString("expvar_port"), profileDuration)
```

Comment:
Though this might be out of scope, I feel like it's only right to bring it up. Should we add facilities to collect the relevant profiles from other agents (trace, process,...) exposing the `pprof` API?  It shouldn't be particularly difficult to at least lay the foundation to perform such collection. For instance, we could make the cpuProfURL parametric here:

```golang
// caller is responsible for providing a valid baseURL ie. fmt.Sprintf("http://127.0.0.1:%s/debug/pprof/profile", config.Datadog.GetString("expvar_port")
// but could also be remote, or an alternate port on a different agent.
func CreatePerformanceProfile(baseURL string, profileDuration int) (*Profile, error) {
	cpuProfURL := fmt.Sprintf("%s?seconds=%d", baseURL, profileDuration)
    // ...
}

```

We might also want to consider an alternative where the profiles are submitted via the API in a push model rather than a pull model. Now this is would 100% be out of scope :)

And of course all of the above can be tackled later. So feel free to ignore.

Reply:
The first one is a quick change and makes sense to me, I'll change it.

The second one would be neat to have, but yeah that one's out of scope. Should I add a card in the backlog or document it elsewhere?

Reply:
@bmikaili I'll create a card - definitely low priority at the moment.